src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -146,6 +165,56 @@ public Command zeroTurretCommand() {
     return Commands.runOnce(() -> this.zeroTurret(), this).withName("ZeroTurretCommand");
   }
 
+  public double calculateTurretPosition() {
+    double encoderPosition =
+        lowGearCANcoder.getPosition().getValueAsDouble()
+            / Constants.TurretConstants.LOW_GEAR_CAN_CODER_RATIO;
+    return encoderPosition + Constants.TurretConstants.ENCODER_ZERO_OFFSET_FROM_TURRET_ZERO_REV;
+  }
+
+  /**
+   * A command to set the current turret position as encoder zero.
+   *
+   * @return a command that sets the current position as encoder zero
+   */
+  public Command zeroZeroing() {
+    return Commands.runOnce(
+        () -> {
+          MagnetSensorConfigs cfg = new MagnetSensorConfigs();
+          cfg.withAbsoluteSensorRange(AbsoluteSensorRangeValue.Unsigned_0To1);
+          cfg.withMagnetOffset(0);
+          CANcoderConfigurator lowGearCANcoderConfigurator = lowGearCANcoder.getConfigurator();
+          CANcoderConfigurator highGearCANcoderConfigurator = highGearCANcoder.getConfigurator();
+          lowGearCANcoderConfigurator.apply(cfg);
+          highGearCANcoderConfigurator.apply(cfg);
+
+          final double trueZeroLowGearOffset =
+              lowGearCANcoder.getAbsolutePosition().getValueAsDouble();
+          final double trueZeroHighGearOffset = highGearCANcoder.getPosition().getValueAsDouble();
```

Comment:
One of these gets Absolute and one gets non-absolute positions. I suspect you want both to be one or the other